File: Java/007.java
import java.util.ArrayList;
import java.util.List;

public class Problem007 {

    /**
     * Find the first x prime numbers.
     */
    public static List<Long> findPrimes(int x) {
        List<Long> primes = new ArrayList<>();
        long y = 2;
        while (primes.size() < x) {
            boolean isPrime = true;
            for (long prime : primes) {
                if (y % prime == 0) {
                    isPrime = false;
                    break;
                }
            }
            if (isPrime) {
                System.out.println("Prime found: " + y);
                primes.add(y);
            }
            y++;
        }
        return primes;
    }

    public static void main(String[] args) {
        int x = 10001;
        List<Long> primes = findPrimes(x);
        System.out.println(x + "st prime: " + primes.get(x - 1));
    }
}

File: Rust/007.rs
fn find_primes(x: usize) -> Vec<u64> {
    let mut primes: Vec<u64> = Vec::new();
    let mut y: u64 = 2;

    while primes.len() < x {
        let mut is_prime = true;
        for &prime in &primes {
            if y % prime == 0 {
                is_prime = false;
                break;
            }
        }
        if is_prime {
            println!("Prime found: {}", y);
            primes.push(y);
        }
        y += 1;
    }
    primes
}

fn main() {
    let x: usize = 10001;
    let primes = find_primes(x);
    println!("{}st prime: {}", x, primes[x - 1]);
}
